### ISSUE TITLE
Use own tokio core

### ISSUE THREAD

**Issue by ondj** [little-dude/rmp-rpc#4]
I want to use your rmp-rpc library with trust_dns. And because `trust_dns` offers `ClientFuture` implementation, I am trying to use async variant. But I think, this is not possible with the current `rmp-rpc` implementation, because it is not possible to get `tokio_core::reactor::Core` handle. 

Probably I need something like this:

```rust
fn main() {
    let mut core = Core::new().unwrap();

    // DNS client
    let dns_address = "8.8.8.8:53".parse().unwrap();
    let (stream, sender) = UdpClientStream::new(dns_address, &core.handle());
    let mut client = ClientFuture::new(stream, sender, &core.handle(), None);

    // MSGPACK RPC Server
    let addr: SocketAddr = "127.0.0.1:54321".parse().unwrap();
    serve(&core.handle(), &addr, &Calculator::new(client))
}
```

Or is there any other possibility?

**Comment by little-dude** [comment]
Ah you're right `serve` should definitely take a `Handle` as argument instead of creating its own core.

**Comment by little-dude** [comment]
@ondj I don't know what you use case is, but this crate has currently a limitation that might disqualify it. I noticed that in some projects using MessagePack-RPC, the clients also have to be able to handle incoming requests and notification from the remote endpoints. With `rmp-rpc`, you cannot do that at the moment: the endpoint is either a client or a server. I'm working on a generic endpoint that could behave both like a client and a server, but I'm not sure when it will be finished: https://github.com/little-dude/rmp-rpc/pull/3

**Comment by ondj** [comment]
No, I just need to query DNS through MSGPACK-RPC server, so current Client is OK.

**Comment by little-dude** [comment]
This should now be fixed in `0.0.10`.
See for example: https://github.com/little-dude/rmp-rpc/blob/master/examples/server.rs#L89:
```rust

fn main() {
    let addr: SocketAddr = "127.0.0.1:54321".parse().unwrap();
    // Create a tokio event loop
    let mut core = Core::new().unwrap();
    // serve returns a future that needs to be spawned on the event loop.
    let server = serve(addr, Echo, core.handle());
    // Run the server. This is blocking. Press ^C to stop
    core.run(server).unwrap();
}
```

**Comment by ondj** [comment]
Great! Now it works as I expected.
